tests/kernel_bundle/kernel_bundle_spec_const.cpp

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "kernel_bundle.h"
+
+namespace kernel_bundle_spec_const {
+
+constexpr sycl::specialization_id<int> SpecName(5);
+constexpr sycl::specialization_id<int> OtherSpecName(5);
+
+TEST_CASE(
+    "Check specialization constants functionality with empty kernel_bundle",
+    "[kernel_bundle]") {
+  sycl::context ctx;
+  const auto always_false_selector = [](auto device_image) { return false; };
+  auto bundle = sycl::get_kernel_bundle<sycl::bundle_state::executable>(
+      ctx, always_false_selector);
+  REQUIRE(bundle.empty());
+
+  CHECK(!bundle.contains_specialization_constants());
+  CHECK(!bundle.native_specialization_constant());
+  CHECK(!bundle.has_specialization_constant<SpecName>());
+}
+
+TEST_CASE(
+    "Check specialization constants functionality with Kernel bundle with "
+    "`kernel_handler::get_specialization_constant()` call",
+    "[kernel_bundle]") {
+  sycl::device device = sycl_cts::util::get_cts_object::device();
+  sycl::queue queue = sycl_cts::util::get_cts_object::queue();
+  sycl::context ctx = queue.get_context();
+
+  using KernelName = class simple_kernel;
+
+  queue.submit([&](sycl::handler &cgh) {
+    cgh.single_task<KernelName>([=](sycl::kernel_handler h) {
+      int a = h.get_specialization_constant<SpecName>();
```

Comment:
No check about the value read?

Reply:
At least add a comment so no one will ask again. :-)
Otherwise, there is no use of `a`, which might trigger some compiler warning.

Reply:
You are right, there is no use of `a`, removed it and add comment.